File: experiments/source/python/inventory_penalty_sum.py
import loom


@loom.entry
def inventory_penalty_sum(target: float, pos: loom.Tensor1) -> float:
    def penalty(pi: float) -> float:
        d = pi - target
        return d * d

    return loom.Tensor1.reduce_sum(loom.Tensor1.map(penalty, pos))

File: experiments/source/cpp/inventory_penalty_sum.cpp
#include <loom/loom.hpp>

LOOM_ENTRY float inventory_penalty_sum(float target, loom::Tensor1 pos) {
  auto penalty = [=](float pi) -> float {
    auto d = pi - target;
    return d * d;
  };
  return loom::Tensor1::reduce_sum(loom::Tensor1::map(penalty, pos));
}
